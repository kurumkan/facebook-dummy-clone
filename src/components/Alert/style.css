.alert {
    background-color: #f2dede;
    color: #a94442;
    border: 1px solid #ebccd1;
    padding: 15px;
    margin-bottom: 20px;
    border-radius: 4px;
}