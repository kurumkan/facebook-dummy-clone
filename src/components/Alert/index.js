import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './style.css';

const Alert = ({ message }) => message ? <div className="alert">{ message }</div> : null;

Alert.propTypes = {
  message: PropTypes.string
};

Alert.defaultProps = {
  message: ''
};

export default Alert;