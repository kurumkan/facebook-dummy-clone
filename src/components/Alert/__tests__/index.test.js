import Alert from '../index.js';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Alert', () => {
  it('renders with error message', () => {
    const rendered = renderer.create(
      <Alert message="error message" />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('does not render', () => {
    const rendered = renderer.create(
      <Alert />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});