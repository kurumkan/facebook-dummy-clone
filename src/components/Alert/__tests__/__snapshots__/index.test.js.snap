// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render 1`] = `null`;

exports[`Alert renders with error message 1`] = `
<div
  className="alert"
>
  error message
</div>
`;
